### ISSUE TITLE
Score, fail-on, fail-under, etc ignores messages emitted by PyLinter itself

### ISSUE THREAD

**Issue by jacobtylerwalls** [pylint-dev/pylint#9059]
### Bug description

Attempt to use `--fail-on=W`, or `evaluation=`, etc., and see they don't take into account messages from PyLinter itself such as `unknown-option-value`.

### Command used

```shell
pylint pylint.lint --enable=unknown --fail-on=W --evaluation='2 if warning else 10'
```

### Pylint output

```python
jwalls@Jacobs-MacBook-Air-4 pylint % pylint pylint.lint --enable=unknown --fail-on=W --evaluation='2 if warning else 10'
************* Module Command line
Command line:1:0: W0012: Unknown option value for '--enable', expected a valid pylint message and got 'unknown' (unknown-option-value)

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 9.98/10, +0.02)

jwalls@Jacobs-MacBook-Air-4 pylint % echo $?
0
```

### Expected behavior

non-zero exit code, score less than 10

### Pylint version

```shell
dev
```

### other
Found while investigating #8457

**Comment by jacobtylerwalls** [comment]
This block doesn't account for the `MSGS` defined at the top of the file.

https://github.com/pylint-dev/pylint/blob/8231e06c9fdf4f12f7dccea7c46ad3d1e626c492/pylint/lint/pylinter.py#L521-L524

**Comment by Pierre-Sassoulas** [comment]
It seems like we should design the order of operations during initialization and configuration loading in a rational and global manner. (I'm working on #5701 atm and this is a mess honestly, there's no way to simply add ``default=x``, because there's so much special cases everywhere. We can't even dream of doing #2293 in the current state either). For the current issue, we should probably iterate on the ``MessageDefinitionStore.messages`` content after proper message initialization but it's not immediately clear to me if there are any special cases related to ``fail-on``/``under`` we need to handle before "proper message initialization" that justify that we did not do that already.
